packages/core-macro/src/component_body/utils.rs

```diff
@@ -0,0 +1,56 @@
+use crate::component_body::ComponentBody;
+use dioxus_core::{Element, Scope};
+use quote::ToTokens;
+use syn::{parse_quote, Path};
+
+/// The output produced by a deserializer.
+///
+/// # For implementors
+/// Struct field guidelines:
+/// * Must be public, so that other deserializers can utilize them.
+/// * Should usually be [`Item`]s that you then simply combine in a [`quote!`]
+/// in the [`ComponentBodyDeserializer::output_to_token_stream2`] function.
+/// * If an [`Item`] might not be included, wrap it in an [`Option`].
+/// * Must be at the component function "level"/"context".
+/// For example, the [`InlinePropsDeserializer`](crate::component_body_deserializers::inline_props::InlinePropsDeserializer)
+/// produces two [`Item`]s; the function but with arguments turned into props, and the props struct.
+/// It does not return any [`Item`]s inside the struct or function.
+pub trait DeserializerOutput: ToTokens {}
```

Comment:
Does this need to be a seperate trait from ToTokens? Are there any specific structural bounds that an implementer must have that make implementing this trait different from ToTokens?

The guidelines here are good, but that can be documented at the module level or when you use this trait

Reply:
> Does this need to be a seperate trait from ToTokens?

Not really, but if I removed `DeserializerOutput`, you could use any struct that implements `ToTokens`, as an output struct. This way, it has to explicitly be a `DeserializerOutput`. And in the future, `DeserializerOutput` might implement more traits or provide a function.

Reply:
> Not really, but if I removed DeserializerOutput, you could use any struct that implements ToTokens, as an output struct. This way, it has to explicitly be a DeserializerOutput.

Unless I am missing something about what `DeserializerOutput` is communicating, I think it should be removed. It seems like the documentation is just talking about how to make something easy to use with quote which is also true for anything that implements ToTokens.

To me this looks similar to adding a blank trait to the built in `Into` trait. You can add extra docs there if you want to, but it makes the code less clear:

```rust
/// Something that can be converted into. To make a struct easy to convert to, you should avoid data that is mixed between multiple fields and unclear names for fields
trait Convertible {}

trait Into<O: Convertible> {
    fn into(self) -> O;
}
```

> in the future, DeserializerOutput might implement more traits or provide a function.

If we need it in the future we could always re-add it.

Reply:
I think it's nice because it has the documentation for it, whereas you can't make documentation for `ToTokens`. I can put the documentation in the module, but having it right in the struct that the `DeserializerArgs::to_output` returns.

Reply:
If you want to keep it, you can make it an automatically implemented by adding:
```rust
impl<T: ToTokens> DeserializerOutput for T {}
```